wpimath/src/test/java/edu/wpi/first/math/kinematics/SwerveDriveKinematicsTest.java

```diff
@@ -406,4 +399,206 @@ void testDesaturateNegativeSpeed() {
+    // - Current tangential velocity: v_t = ω × r = 2π × 16.97 = 106.63 m/s
+    //
+    // Two acceleration components:
+    // 1) Tangential acceleration (from angular acceleration α = 2π rad/s²):
+    //    a_tangential = α × r = 2π × 16.97 = 106.63 m/s²
+    //    Direction: perpendicular to radius vector
+    //
+    // 2) Centripetal acceleration (from angular velocity ω = 2π rad/s):
+    //    a_centripetal = ω² × r = (2π)² × 16.97 = 668.7 m/s²
+    //    Direction: toward center of rotation
+    //
+    // Total acceleration magnitude: |a_total| = √(a_tangential² + a_centripetal²)
+    //                                         = √(106.63² + 668.7²) = 678.4 m/s²
+    //
+    // For module positions:
+    // FL (12, 12):   radius angle = 135°, tangential = 45°,  centripetal = -135° → total angle =
+    // -144°
+    // FR (12, -12):  radius angle = 45°,  tangential = -45°, centripetal = -135° → total angle =
+    // 126°
+    // BL (-12, 12):  radius angle = 135°, tangential = 45°,  centripetal = 45°   → total angle =
+    // -54°
+    // BR (-12, -12): radius angle = -45°, tangential = 45°,  centripetal = 135°  → total angle =
+    // 36°
+    //
+    // The acceleration angles are not simply tangential because the centripetal component
+    // from the existing angular velocity dominates and affects the direction.
+
+    double centerRadius = m_kinematics.getModules()[0].getNorm();
+    double tangentialAccel = centerRadius * accelerations.alpha; // α * r
+    double centripetalAccel = centerRadius * angularVelocity * angularVelocity; // ω² * r
+    double totalAccel = Math.hypot(tangentialAccel, centripetalAccel);
+
+    Rotation2d[] expectedAngles =
+        Arrays.stream(m_kinematics.getModules())
+            .map(
+                m -> {
+                  Rotation2d radiusAngle = m.getAngle();
+
+                  // Tangential acceleration: perpendicular to radius (90° CCW from radius)
+                  Rotation2d tangentialDirection = radiusAngle.rotateBy(Rotation2d.kCCW_Pi_2);
+                  double tangentialX = tangentialAccel * tangentialDirection.getCos();
+                  double tangentialY = tangentialAccel * tangentialDirection.getSin();
+
+                  // Centripetal acceleration: toward center (opposite of radius)
+                  Rotation2d centripetalDirection = radiusAngle.rotateBy(Rotation2d.kPi);
+                  double centripetalX = centripetalAccel * centripetalDirection.getCos();
+                  double centripetalY = centripetalAccel * centripetalDirection.getSin();
+
+                  // Vector sum of tangential and centripetal accelerations
+                  double totalX = tangentialX + centripetalX;
+                  double totalY = tangentialY + centripetalY;
+
+                  return new Rotation2d(totalX, totalY);
+                })
+            .toArray(Rotation2d[]::new);
+
+    assertAll(
+        () -> assertEquals(totalAccel, moduleAccelerations[0].acceleration, 0.1),
```

Comment:
Could the tolerances be changed from 0.1 to `kEpsilon`? I'm not sure why some of the other tests use 0.1 instead of `kEpsilon`, but it'd be good to specify a higher level of tolerance.

Reply:
Does lowering the alpha tolerance in testTurnInPlaceForwardAccelerations work locally?